testplan/common/entity/base.py

```diff
@@ -269,7 +295,7 @@ def stop_in_pool(self, pool, reversed=False):
                 continue
             else:
                 resource.wait(resource.STATUS.STOPPED)
-                resource.logger.debug("Stopped %r", resource)
```

Comment:
Agree on that, however, the formatting this way would be 100% aligned eventually and the speed up is ~100 times. Two points that may be important for context: do we log to external file or to stdout by default and what is the level of control the user has over logging level?

Reply:
Using lazy-logging is pylint recommended and considered good style - so I would rather that we use %-style for logging consistently in our code base not the other way around. With that said, I'm aware that we almost always enable debug logging when we writing to testplan.log, so that won't have much actual benefit but it's still good to stick with good practice. btw, where does the ~100 times speed-up come from?

Reply:
You were right to guess, I looked at the performance in iPython yesterday (timeit is more convenient there), but in CPython, it is less convincing (letdown...), 1.5-2 times. In iPython, this is 20-100 times dependent on types and size.

In principle, the speed up would come from f-string making more of the bytecode compiler. Let me cite one of the Python core developers here: https://twitter.com/SerhiyStorchaka/status/1206127524218130432?ref_src=twsrc%5Etfw
"
1. It can calculate the length and the kind of the resulting string and allocate a memory only once.
2. It do not parse the format string at run time.
3. It do not create a tuple, all values are on the stack.
4. It do not look up an attribute and call a method.
"
This of course, already compares with .format method more.

Let me step back on the f-strings then and go for the recommended style.